llms/mlx_lm/models/base.py

```diff
@@ -60,3 +55,26 @@ def from_dict(cls, params):
                 if k in inspect.signature(cls).parameters
             }
         )
+
+
+def create_additive_causal_mask(N: int, offset: int = 0):
+    if offset > 0:
+        rinds = mx.arange(offset + N)
+        linds = mx.arange(offset, offset + N) if offset else rinds
+        mask = linds[:, None] < rinds[None]
+        return mask * -1e9
+    return nn.MultiHeadAttention.create_additive_causal_mask(N)
```

Comment:
A simpler alternative for this function would be to slice the output of `nn.MultiHeadAttention.create_additive_causal_mask`:

```
def create_additive_causal_mask(N: int, offset: int = 0):
    return nn.MultiHeadAttention.create_additive_causal_mask(N)[offset:,]
```

I'm not sure why the original author chose to create the offset mask from `aranges`, so to minimize change/risk I'm keeping the original code when there's an offset.